packages/flutter/lib/src/widgets/media_query.dart

```diff
@@ -787,6 +788,32 @@ class MediaQuery extends InheritedWidget {
     );
   }
 
+  /// Creates a [_MediaQueryFromWindow] which builds and updates a
+  /// [MediaQuery] using the latest [WidgetsBinding.window] values.
+  ///
+  /// The [MediaQuery] is wrapped in a separate widget to ensure that only it
+  /// and its dependents are updated when `window` changes, instead of
+  /// rebuilding the whole widget tree.
+  ///
+  /// This should be inserted into the widget tree when the [MediaQuery] view
+  /// padding is consumed by a widget in such a way that the view padding is no
+  /// longer exposed to the widget's descendants or siblings.
+  ///
+  /// The [child] argument is required and must not be null.
+  ///
+  /// See also:
+  ///
+  ///  * [_MediaQueryFromWindow], the underlying widget.
+  static Widget fromWindow({
```

Comment:
I think we should keep this change. This is valuable to folks who may want it. :)